exec/java-exec/src/main/java/org/apache/drill/exec/store/dfs/FileSelection.java

```diff
@@ -47,16 +47,25 @@
   private List<FileStatus> statuses;
 
   public List<String> files;
+  /**
+   * root path for the selections
+   */
   public final String selectionRoot;
+  /**
+   * root path for the metadata cache file (if any)
+   */
+  public final String cacheFileRoot;
```

Comment:
That was my initial approach (updating the selectionRoot without keeping a separate cacheFileRoot).   However,  I ran into a few issues.  The main one that I recall is that the dir0, dir1 etc columns are associated with the selectionRoot, so suppose I run the following query: 
    SELECT dir0, dir1 FROM dfs.tmp.t2 WHERE dir0=2015 AND dir1='Q1' 
and if the selectionRoot gets modified to point to '2015/Q1'  then we have lost the context of the original dir0, dir1 because everything will become relative to the new selectionRoot.    This produces wrong results.   The same problem occurred with a SELECT *  query where the directory columns where not showing up correctly.